samples/test-samples.cmake

```diff
@@ -1,14 +1,20 @@
 # Copyright (c) Open Enclave SDK contributors.
 # Licensed under the MIT License.
 
-# This script requires the variables SOURCE_DIR, BUILD_DIR, and
+# This script requires the variables SOURCE_DIR, BUILD_DIR,
+# COMPILER_SUPPORTS_SNMALOC, USE_DEBUG_MALLOC and
 # PREFIX_DIR to be defined:
 #
 #     cmake -DHAS_QUOTE_PROVIDER=ON -DSOURCE_DIR=~/openenclave -DBUILD_DIR=~/openenclave/build -DPREFIX_DIR=/opt/openenclave -P ~/openenclave/samples/test-samples.cmake
 
 # Set SAMPLES_LIST so that helloworld becomes the first if BUILD_ENCLAVES=ON.
 if (BUILD_ENCLAVES)
   set(SAMPLES_LIST helloworld file-encryptor switchless host_verify)
+  # Debug malloc will set allocated memory to a fixed pattern.
+  # Hence do no enable pluggable_allocator test under USE_DEBUG_MALLOC.
```

Comment:
I don't follow this, debug malloc at least used to work with snmalloc (specific changes were made in #2649 for that to be the case), what has changed that means it no longer does?

Reply:
debug malloc still works with snmalloc. However since debug malloc sets each byte of the allocated object, it can slow down the application enough to cause test timeout, especially in tests like this one that perform a large number of allocation.